initial version for discussion

amitai-cet opened the pull request.

amitai-cet committed:
initial version for discussion

UmanShahzad commented on cetloxblock/docs/TipsAndTricks.html:
```diff
@@ -0,0 +1,106 @@
+    <li>API reference: <a target="_blank" href="http://edx.readthedocs.io/projects/xblock/en/latest/index.html">http://edx.readthedocs.io/projects/xblock/en/latest/index.html </a></li>
+    <li><a target="_blank" href="http://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/index.html">http://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/index.html </a></li>
+    <li>ההנחיות להתקנת virtualenv ב-windows: <a target="_blank" href="http://timmyreilly.azurewebsites.net/python-pip-virtualenv-installation-on-windows/">http://timmyreilly.azurewebsites.net/python-pip-virtualenv-installation-on-windows/ </a></li>
+    <li>Antother tutorial: <a target="_blank" href="http://opencraft.com/doc/edx/xblock/tutorial.html">http://opencraft.com/doc/edx/xblock/tutorial.html </a></li>
```
Nit: `Antother` -> `Another`

UmanShahzad commented on cetloxblock/setup.py:
```diff
@@ -0,0 +1,41 @@
+
+    All of the files under each of the `roots` will be declared as package
+    data for package `pkg`.
+
```
Nit: extra newline.

UmanShahzad commented on cetloxblock/setup.py:
```diff
@@ -0,0 +1,41 @@
+setup(
+    name='cetloxblock-xblock',
+    version='1.0.0',
+    description='cetloxblock XBlock',   # TODO: write a better description.
```
You can probably just use a re-worded version of the `Goal` section in your `README.md`:

"An XBlock that embeds the CET Player platform into Open edX. The CET player contains the ability to create personlized learning path for each student."

UmanShahzad replied:
In addition, you can add a `long_description` which contains the entire `README.md`. Example: https://github.com/edx/edx-enterprise/blob/master/setup.py#L43 -- notice how they grab the README file pretty easily and just stick it there.

amitai-cet replied:
Implemented.
For this I removed the upper level directory, to have the readme file in the same level as the setup.py. This corresponds to your [1st (closed) pull request](https://github.com/ILCET/lo-activity-xblock/pull/1).

But where can I see the long_description in action?

pomegranited replied:
The information in setup.py forms the package information provided if you package and distribute your code.  This is optional, since we can install directly from github, but some packages choose to do this.

For the edx-enterprise example that @UmanShahzad  gave, they distribute that package to https://pypi.org/project/edx-enterprise/, and so the long description becomes the package information shown there.

More info: http://python-packaging.readthedocs.io/en/latest/metadata.html

UmanShahzad commented on cetloxblock/cetloxblock_xblock.egg-info/entry_points.txt:
```diff
@@ -0,0 +1,3 @@
+[xblock.v1]
```
We need to add a `.gitignore` file to ignore these things in version control -- they shouldn't be in the public repository.

The same goes for `.pyc` files, which seemed to have gotten checked in.

Here's an example you can take and drop in the root of this repo: https://github.com/open-craft/xblock-poll/blob/master/.gitignore

UmanShahzad replied:
Also, since I'm here, I should say we need quality checks on this code to keep to a standard as we develop.

For this, we'll need a `requirements.txt` file, similar to what you see [here](https://github.com/open-craft/problem-builder/blob/master/requirements.txt). (Only include what we need, of course). We can run `pylint` or `flake8`, whatever you prefer, but that'll need to be added to the `requirements.txt` file, and pinned to a version.

Then we can enforce that quality checks using that tool pass before merging any new PRs in the future.

amitai-cet replied:
Sorry, I need help with this comment. What should I have in requirements.txt? my single .py file imports only pkg_resources, and setup.py imports os. Should they be there? I am not familiar with python and its testing tools. I install my xblock to devstack without requirements.txt.

pomegranited replied:
In the XBlock development guide we shared, under [Coding standards](https://github.com/open-craft/doc/blob/master/docs/xblock-dev-guidelines.md#coding-standards), it discusses the `requirements.txt` and `pylint` tool @UmanShahzad  referred to.

For example, to run pylint over your code, we'd need to install pylint as a dependency of your XBlock.

So your `requirements.txt` file might contain:
```
pylint==1.4.5
```

And to get pylint, we'd install it like this:
```bash
pip install -r requirements.txt
```

UmanShahzad commented on cetloxblock/cetloxblock/static/html/cetlo_studio_view.html:
```diff
@@ -0,0 +1,25 @@
+    </ul>
+  </div>
+</div>
+
```
Nit: extra newline.

UmanShahzad commented on cetloxblock/cetloxblock/static/html/cetlo_studio_item_template.html:
```diff
@@ -0,0 +1,7 @@
+    <div class="cet-lo-item-title"><%=title%></div>
+    <div class="cet-lo-item-view"><a class="cet-lo-item-link" href="<%=link%>" target="_blank">view</a></div>
+  </li>
+</script>
```
Nit: this needs an extra newline.

UmanShahzad commented on cetloxblock/cetloxblock/static/html/cetlo_student_view.html:
```diff
@@ -0,0 +1,11 @@
+
+<div class="cetloxblock-container">
+  <iframe class="cet-lo-player" border="0" src="{iframeSrc}"></iframe>
+</div>
```
Nit: this needs an extra newline.

UmanShahzad commented on cetloxblock/cetloxblock/static/css/cetloxblock_student.css:
```diff
@@ -0,0 +1,11 @@
+.cet-lo-player {
+  width: 100%;
+  height: 800px;
+}
```
Nit: this needs an extra newline.

UmanShahzad commented on cetloxblock/cetloxblock.sln:
```diff
@@ -0,0 +1,28 @@
+﻿
```
When you make a `.gitignore` file, this should also be ignored if it isn't already, i.e. `*.sln` files.

UmanShahzad commented on cetloxblock/cetloxblock.pyproj.vspscc:
```diff
@@ -0,0 +1,10 @@
+﻿""
```
When you make a `.gitignore` file, this should also be ignored if it isn't already, i.e. `*.vspscc` files.

UmanShahzad commented on cetloxblock/cetloxblock.vssscc:
```diff
@@ -0,0 +1,10 @@
+﻿""
```
When you make a `.gitignore` file, this should also be ignored if it isn't already, i.e. `*.vssscc` files.

UmanShahzad commented on cetloxblock/cetloxblock.pyproj.user:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
When you make a `.gitignore` file, this should also be ignored if it isn't already, i.e. `*.pyproj.*` files.

UmanShahzad commented on cetloxblock/cetloxblock.pyproj:
```diff
@@ -0,0 +1,66 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
When you make a `.gitignore` file, this should also be ignored if it isn't already, i.e. `*.pyproj` files.

UmanShahzad commented on cetloxblock/CetInsights/CetInsightsPage.html:
```diff
@@ -0,0 +1,83 @@
+﻿<!--Add the following HTML code to CET Insights page's RAW html-->
+<script src="https://login.cet.ac.il/Scripts/login.js"></script>
+<script src="//ebag.dev.cet.ac.il/CetSso.js"></script>
```
Cool relative protocol :) But as the notice [in this article](https://www.paulirish.com/2010/the-protocol-relative-url/) says, it's probably good to be explicit about the protocol, i.e. `https://` rather than `//`.

UmanShahzad replied:
Also, I tried loading http://ebag.dev.cet.ac.il/CetSso.js or https://ebag.dev.cet.ac.il/CetSso.js, and it doesn't work for me.

amitai-cet replied:
This whole file was not supposed to be a part of the component and will be deleted.

UmanShahzad commented on cetloxblock/CetInsights/CetInsightsPage.html:
```diff
@@ -0,0 +1,83 @@
+    CET_DOMAIN = ".dev.cet.ac.il"
+  }
+  var PRODUCT_PLAYER = "//productplayer" + CET_DOMAIN + "/";
+  var PRODUCT_PLAYER_SERVICES = "//productplayerservices" + CET_DOMAIN + "/";
```
Similar recommendation as above -- we should be explicit about the protocol.

UmanShahzad commented on cetloxblock/CetInsights/CetInsightsPage.html:
```diff
@@ -0,0 +1,83 @@
+  if (window.location.host.indexOf("localhost") >= 0 || window.location.host.indexOf("127.0.0.1") >= 0) {
+    CET_DOMAIN = ".dev.cet.ac.il"
+  }
+  var PRODUCT_PLAYER = "//productplayer" + CET_DOMAIN + "/";
```
Similar recommendation as above -- we should be explicit about the protocol.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+  """
+
+  CET_DOMAIN = u".lab.cet.ac.il"
+  ITEM_URL_TEMPLATE = u"//lo{cetdomain}/player/?document={documentid}&language={language}&sitekey=ebag#options=nobar"
```
Similar recommendation as above -- we should be explicit about the protocol.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+
+  CET_DOMAIN = u".lab.cet.ac.il"
+  ITEM_URL_TEMPLATE = u"//lo{cetdomain}/player/?document={documentid}&language={language}&sitekey=ebag#options=nobar"
+  TASK_URL_TEMPLATE = u"//lo{cetdomain}/player/xblockplayer.html?task={taskid}&sitekey=ebag"
```
Similar recommendation as above -- we should be explicit about the protocol.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+  CET_DOMAIN = u".lab.cet.ac.il"
+  ITEM_URL_TEMPLATE = u"//lo{cetdomain}/player/?document={documentid}&language={language}&sitekey=ebag#options=nobar"
+  TASK_URL_TEMPLATE = u"//lo{cetdomain}/player/xblockplayer.html?task={taskid}&sitekey=ebag"
+  #TASK_URL_TEMPLATE = u"//lo{cetdomain}/player/?task={taskid}&sitekey=ebag#options=nobar"
```
In the official version of this XBlock (I suppose this PR is a part of making it official and production-ready), we should really keep all commented-out code out of the repository.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+    The studio view of the CetLoXBlock, letting author to select a LO.
+    This is a stub showing a fix list of LO items.
+    """
+
```
Nit: extra newline.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+    self.documentid = data["documentid"].decode('utf8')
+    self.language = data["language"].decode('utf8')
+    self.display_name = data["title"]
+
```
Nit: extra newline. Doesn't really help readability to keep this separate IMO.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+  def add_sso_scripts(self, fragment):
+    """ add scripts required for sso with CET """
+    fragment.add_javascript_url(u"https://login.cet.ac.il/Scripts/login.js")
+    fragment.add_javascript_url(u"//ebag{cetdomain}/CetSso.js".format(cetdomain=self.CET_DOMAIN))
```
Similar recommendation as above -- we should be explicit about the protocol.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+    data = data.decode("utf8")
+    return data
+
+  has_author_view = True
```
It would probably be better to move this up somewhere, so it's not in between functions, to keep things organized.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+import pkg_resources
+#import urllib
+from xblock.core import XBlock
+#from xblock.fields import String, Scope, ScopeBase, UserScope, BlockScope
```
Similar comment as above -- we should keep commented-out code out of the repository, until it's actually needed.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+"""
+
+import pkg_resources
+#import urllib
```
Similar comment as above -- we should keep commented-out code out of the repository, until it's actually needed.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+  # ALL STRINGS ARE UNICODE!
+  # ALL FUNCTION DEFS GET UNICODE PARAMETERS AND RETURN UNICODE VALUES!
+  # KEEP THIS SO FOR YOUR HEALTH AND WELL BEING.
+  #############################################################################
```
Is this actually needed if using the correct Python version? I don't see that this has been required for many other XBlocks, and they aren't calling `decode` anywere. I may have misunderstood something, so do clarify.

amitai-cet replied:
I just followed the [tutorial](http://edx.readthedocs.io/projects/xblock-tutorial/en/latest/getting_started/prereqs.html) and installed python 2.7. Which version do you recommend for me to install?

pomegranited replied:
Yep, you're correct, Open edX runs python 2.7.  Just not sure why you need to `decode` all the strings?

amitai-cet replied:
Only because otherwise I get errors regarding enoding...

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+  # KEEP THIS SO FOR YOUR HEALTH AND WELL BEING.
+  #############################################################################
+
+  # EDX field: Component's display name in studio and course
```
XBlock developers will be able to tell if the field is overridden or not, and IDEs will help with this if not. So we should be good to remove this comment as it will loom around unhelpfully otherwise.

UmanShahzad commented on cetloxblock/cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
```
I noticed we're using 2-space tabs in the Python files here. That's OK, but most people's IDEs are probably configured to use 4-space as that's the norm throughout the community, and what the majority by far is used to.

Not forcing anything here, but I do suggest to re-tab the Python code to be 4-space. Eventually this will become important for quality checking tools too.

If you really want to use 2-space though, then when we configure a quality checking tool like `pep8` or `pylint`, we need to configure it to accept this 2-spacing.

amitai-cet replied:
tabs changed to 4-space.

UmanShahzad commented on cetloxblock/CetInsights/CetInsightsPage.html:
```diff
@@ -0,0 +1,83 @@
+
+  function setLink(courseCetName) {
+    var link = document.getElementById('cet-insights-link');
+    link.href = "//productplayer" + CET_DOMAIN + "/" + courseCetName;
```
Similar recommendation as above -- we should be explicit about the protocol.

Also we should use `PRODUCT_PLAYER + courseCetName` here I think since we defined it above.

UmanShahzad commented on cetloxblock/CetInsights/CetInsightsPage.html:
```diff
@@ -0,0 +1,83 @@
+</div>
+<div id="cet-insights-student" style="display:none;text-align:center;direction:rtl;">
+  לא ניתן לצפות במערכת הדוחות של מטח כרגע, יש  לפנות למורה.
+</div>
```
Nit: needs extra newline.

UmanShahzad commented on cetloxblock/docs/TipsAndTricks.html:
```diff
@@ -0,0 +1,106 @@
+  deactivate <br />
+
+</body>
+</html>
```
Nit: needs extra newline.

amitai-cet replied:
This whole file was not supposed to be a part of the component and will be deleted.

UmanShahzad reviewed (changes requested):
Thanks @amitai-cet ! These requests for changes are mostly cosmetic right now, but it's a big step in moving towards having a very high-quality XBlock.

I'll continue to review the functionality once we move forward with this iteration.

Thanks for the work so far!

amitai-cet committed:
Deleting and ignoring setup and IDE files

amitai-cet committed:
implement code review: long desc, explicit https protocol, some cosmetics

amitai-cet committed:
remove extra directory root level

amitai-cet committed:
dynamic detection of cet environment and domain

amitai-cet commented:
Hi Uman / Jill!
All comments implemented, except of 2 for which I requested clarifications (requirements.txt & Python version).
Please look at my replies, which are present only when I had something to clarify or to ask.
In addition, I moved all files one level up, and committed a last change concerning dynamic url construction.
Thank you for the support!

pomegranited commented on setup.py:
```diff
@@ -0,0 +1,46 @@
+    return {pkg: data}
+
+base_path = os.path.dirname(__file__)
+README = open(os.path.join(base_path, "README.md")).read()
```
To let your README file be included in your package so it can be read here, you'll need to add a `MANIFEST.in` file: http://python-packaging.readthedocs.io/en/latest/non-code-files.html

You can see the error if you try to install from the distribution package:
```bash
edxapp@vagrant:/edx/src/lo-activity-xblock$ python setup.py sdist
edxapp@vagrant:/edx/src/lo-activity-xblock$ pip install dist/cetloxblock-xblock-1.0.1.tar.gz 
Processing ./dist/cetloxblock-xblock-1.0.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-4HFE_3-build/setup.py", line 24, in <module>
        README = open(os.path.join(base_path, "README.md")).read()
    IOError: [Errno 2] No such file or directory: '/tmp/pip-4HFE_3-build/README.md'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-4HFE_3-build/
```

amitai-cet replied:
Fine.
I understand the error, though I wonder why I don't get it when installing to devstack.

pomegranited replied:
When you install using just `pip install` or `pip install -e` from the main source directory, then it doesn't run `setup.py` and create a package, so it doesn't hit the error.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+  var PRODUCT_PLAYER_API = "//productplayer" + CetDomain + "/api/ApiProxy/";
+  var PRODUCT_PLAYER_SERVICES = "//productplayerservices" + CetDomain + "/";
+  var TREE_SERVICE = "//treeservice" + CetDomain + "/api/Trees/";
+  var TREE_LMS_SERVICE = "//treeservice" + CetDomain + "/api/TreeLMS/";
```
[As noted previously](https://github.com/ILCET/lo-activity-xblock/pull/2#discussion_r180068384), all of these URLs should use the `https` protocol explicitly.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+    if (match.length) {
+      return match[1];
+    }
+  }
```
I'm pretty sure you can get the current course ID from the DOM, [like we do here](https://github.com/open-craft/problem-builder/blob/fc465870ef67b152fe4f60d7b9865f8748911a90/problem_builder/public/js/instructor_tool.js#L263).

amitai-cet replied:
This was very helpful. I expected data of this type to be documented somewhere (which unfortunately I did not find), so I know I can count on it. Nevertheless the url format is hardly safer.

amitai-cet replied:
Your suggestion is working in the campus, but fails on devstack (the MASTER that I installed when I started the project). Feeling it a bit shaky, I left the previous algorithm as a fallback.

pomegranited replied:
FYI, if you develop using the [Campus devstack](https://gitlab.com/opencraft/client/campus/campus-devstack) (which runs [`opencraft-release/ginkgo.1-campus`](https://github.com/open-craft/edx-platform/tree/opencraft-release/ginkgo.1-campus)), then you'll find it easier to debug issues on deployment.

Also note that edX have [removed the vagrant devstack in favour of Docker](https://github.com/edx/configuration/pull/4492), so if you have to rebuild your `master` stack, you may run into issues.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+          dfd.resolve(courseObj.path);
+        }
+        else {
+          //dfd.reject("Cet course is not defined");
```
Please remove commented-out code.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+          dfd.resolve(DEFAULT_CET_COURSE);
+        }
+      }).fail(function (response) {
+        //dfd.reject("Failed to retrieve Cet course name. Error: " + response);
```
Please remove commented-out code.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+    }
+
+
+    // http://treeservice.cet.ac.il/api/Trees/a2326f9f-c094-4458-95ea-e832d7273e9c/he
```
These example URLs are really useful for me to understand the data this code is processing!  Thanks for including them.

pomegranited commented on cetloxblock/static/js/services.js:
```diff
@@ -0,0 +1,202 @@
+  var CetDomain = this.CetDomain = this.getCetDomain();
+
+  var PRODUCT_PLAYER = "//productplayer" + CetDomain + "/";
+  var PRODUCT_PLAYER_API = "//productplayer" + CetDomain + "/api/ApiProxy/";
```
Could reduce some copy-paste by doing this:
```javascript
var PRODUCT_PLAYER = "//productplayer" + CetDomain + "/";
var PRODUCT_PLAYER_API = PRODUCT_PLAYER + "api/ApiProxy/";
```

pomegranited reviewed (commented):
Thanks for making those changes @amitai-cet !  The code is getting much cleaner.  I've replied to the questions I could find, but haven't found where you asked about the "python version"?

* Please add a `.gitignore` file like [Uman
  described](https://github.com/ILCET/lo-activity-xblock/pull/2/files/cc77da9a012a8a20fee7abef2fe27ee19f0f4993#r180066103) -- that will help avoid accidentally committing unwanted files in the future.
* Will need to add tests eventually too, cf
  [Testing](https://github.com/open-craft/doc/blob/master/docs/xblock-dev-guidelines.md#testing)

amitai-cet commented:
Thank you for the helpful comments. I must make some points (in addition to replies in-place):

1. My question about python version is [here (line 29 in cetloxblock.py in pull requset)](https://github.com/ILCET/lo-activity-xblock/pull/2/files/cc77da9a012a8a20fee7abef2fe27ee19f0f4993#diff-e2e13d985e6c6aaefe753c41b33b5f2bR29)
2. I added in my local repo directory the MANIFEST.in and requirements.txt, but then they were automatically ignored by git, exactly as the .gitignore file itself. I added all of them to the commit to satisfy your comments.
3. I understand the requirements of pylint and unit test. I don't get into it right now, there is some learning I have to do in advance.

So far for the code review. Another commit is coming soon including bug fixes.
BTW: Should we proceed with the current pull request, ot should we close it and open a new one? Which way will be easier to track the inline comments?

amitai-cet committed:
more explicit protocol in urls, add manifest.in and requirements.txt, detect course name using DOM

amitai-cet committed:
add folder to lo attributes, override default sso/idm environment in cet

pomegranited commented on cetloxblock/cetloxblock.py:
```diff
@@ -0,0 +1,138 @@
+
+        frag = Fragment(html)
+        frag.add_css(self.resource_string("static/css/cetloxblock_studio.css"))
+        frag.add_javascript(self.resource_string("static/js/studio-xblock.js"))
```
Missing these JS files when editing in Studio:
```python
frag.add_javascript(self.resource_string("static/js/env-manager.js"))
frag.add_javascript(self.resource_string("static/js/sso.js"))
```

amitai-cet replied:
Right. 
The 1st one added. I don't see need for sso.js in studio view.

pomegranited reviewed (approved):
@amitai-cet I've followed up on your [python question](https://github.com/ILCET/lo-activity-xblock/pull/2/files/cc77da9a012a8a20fee7abef2fe27ee19f0f4993#diff-e2e13d985e6c6aaefe753c41b33b5f2bR29).

And have left one comment about missing JS files in the Studio Edit view.

But once that's addressed, and you're happy with this initial version, you're welcome to merge this PR and start a new one.  Usually, each PR has a stated purpose, e.g. to add a feature, to fix a bug, or to add some other capability.  So if you're ready to release this version to move on to the next steps, feel free to merge.

Also, if you choose "Squash and Merge", then your revision history gets squashed down to just a single commit, which is nice when there's lots of commits that you don't need to keep.

After we merge, we often tag our releases to match the number used in [setup.py](https://github.com/ILCET/lo-activity-xblock/blob/initial-version/setup.py#L28), so we can easily see a progression for what versions are released where/when.  For example, after you merge this PR, you can then run these steps to create a tag:
```bash
lo-activity-xblock/ $ git checkout master
lo-activity-xblock/ $ git pull
lo-activity-xblock/ $ git tag v1.0.10
lo-activity-xblock/ $ git push origin v1.0.10
```

This will make the tag `v1.0.10` show up on github under the Branch > Tags drop down.

Then, I'll be able to install it on campus-dev.opencraft.hosting with this command:
```
pip install git+https://github.com/ILCET/lo-activity-xblock.git@v1.0.10#egg=lo-activity-xblock==v1.0.10
```

Instead of using the explicit hash references of your latest commit, like I do now:
```
pip install git+https://github.com/ILCET/lo-activity-xblock.git@26103df1651b09962e83bf8781ad54b6a0b6e06b#egg=lo-activity-xblock
```

The result is the same, it just looks cleaner, and helps to keep track of version sequence.

:+1: once my comment is addressed.

- [x] I tested this on my Campus (ginkgo) devstack:
  * I've confirmed that the xblock can be packaged and installed
  * I've confirmed that the xblock loads without error in Studio and LMS
    However, it shows an empty page, so maybe my host isn't allowed to connect to the CET Player platform?
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> -- see above re blank screen.
- [x] Includes documentation.

amitai-cet commented:
I added the missing script in studio and replied the python comment.
The blank screen you see is because we don't have yet a trusted SSL certificate for our TESTING sites. It is being handled by our IT in those days. For the meantime I have to confirm manually the untrusted sites. Then, as I wrote before, in this version we can let in only users registered in CET. Only we try to authenticate them through EDU sign-in present in your staging campus.

I want to try to solve our EDU authentication problem before announcing a release, so please install the xblock from the branch once more.
I learn much from your comments. Primarily I develop for windows in Visual Studio...
Thank you for all

amitai-cet committed:
add missing file reference to studio view

pomegranited commented:
@amitai-cet  I'm so glad this process is helping!  If you have any other tips or lessons learned that aren't already in our [XBlock developer guidelines](https://github.com/open-craft/doc/blob/master/docs/xblock-dev-guidelines.md), would you be willing to submit a PR to update them?

Deploying your latest version to the staging server now.

amitai-cet committed:
complete IDM authentication to CET

amitai-cet committed:
version number
